### PR TITLE
Update tracking structures during log deletion

### PR DESCRIPTION
**Pull request by morgando**
The changes in this PR delete from the checkpoint list and the timestamp-to-lsn mapping structure when a logfile is deleted if modsnap is enabled.

### CODE REVIEW

File: bdb/file.c [hunk 1]
```diff
@@ -3870,22 +3877,6 @@ static void delete_log_files_int(bdb_state_type *bdb_state)
...
-                    if (bdb_state->attr->debug_log_deletion)
-                        logmsg(LOGMSG_USER, "not ok to delete log, log file needed "
-                                        "to recover to at least %ds ago\n",
-                                bdb_state->attr->min_keep_logs_age);
-                    if (ctrace_info)
-                        ctrace("not ok to delete log, log file needed to "
-                               "recover to at least %ds ago\n",
-                               bdb_state->attr->min_keep_logs_age);
-                    break;
-                }
```
Comment (morgando) [inline bdb/file.c:3887]: The above block was duplicated. Consolidated to block following line 3900

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>8/586 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>unionpar_maxqueue [setup failure]<br>leakcheck [setup failure]<br>basic_ssl_generated [setup failure]<br>basic_nogenid48_generated [setup failure]<br>snapshot_during_truncate<br>timepart_trunc<br>snap_ha_retry_newsi_generated<br>dbcreate</p>

**Review by riverszhang89 — APPROVED** [review]

**Comment by morgando** [comment]
`cdb2test   Nov  6 22:48:53 2024  in-comdb2db modsnap_logdel.R20241106.12`

**Comment by morgando** [comment]
Thanks Rivers!

**Pull request merged**